Register login logout

Nothemba opened the pull request.

lindokuhlendamane25 committed:
added register login logout routes

lindokuhlendamane25 committed:
added  home route

lindokuhlendamane25 committed:
modified user

jens1101 reviewed (changes requested):
The `node_modules` folder is still committed in spite of being in the gitignore file. Please run `git rm -r --cached node_modules` and commit in order to remove it. See [this SO answer](https://stackoverflow.com/a/62389965/1358008) for details.

lindokuhlendamane25 committed:
removed node modules

Nothemba commented:
Removed node modules, and also added a git ignore file on the backend
folder.

On Wed, 27 Jan 2021, 12:53 Jens Tschirpig, <notifications@github.com> wrote:

> *@jens1101* requested changes on this pull request.
>
> The node_modules folder is still committed in spite of being in the
> gitignore file. Please run git rm -r --cached node_modules and commit in
> order to remove it. See this SO answer
> <https://stackoverflow.com/a/62389965/1358008> for details.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jens1101/moody-socials/pull/2#pullrequestreview-577211872>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFQNGY4VMZDASC2LUYRMHI3S37WC7ANCNFSM4WU4FKFA>
> .
>

jens1101 commented on backend/routes/pages.js:
```diff
@@ -0,0 +1,82 @@
+    let user = req.session.user;
+
+    if(user) {
+        res.render('home', {opp:req.session.opp, name:user.username});
```
- Is a render necessary here? I believe you could just respond with JSON here.
- I cannot find any view named "home" in you code, so I don't believe this render will work at all.

jens1101 reviewed (changes requested):
- `mysql` and `pug` are not used anywhere in this code. Why are they dependencies?
- When I run `npm start` I receive the following error:
  ```
  > moody-socials-backend@1.0.0 start /Users/jens/Documents/Projects/moody-socials/backend
  > cd server && nodemon server.js
  
  sh: line 0: cd: server: No such file or directory
  npm ERR! code ELIFECYCLE
  npm ERR! errno 1
  npm ERR! moody-socials-backend@1.0.0 start: `cd server && nodemon server.js`
  npm ERR! Exit status 1
  npm ERR! 
  npm ERR! Failed at the moody-socials-backend@1.0.0 start script.
  npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  
  npm ERR! A complete log of this run can be found in:
  npm ERR!     /Users/jens/.npm/_logs/2021-01-27T11_47_30_747Z-debug.log
  
  Process finished with exit code 1
  ```

lindokuhlendamane25 committed:
removed unneccessary dependancies

Nothemba commented:
I've changed the startup script it works now, removed those dependencies
and unnecessary code.

On Wed, 27 Jan 2021, 13:52 Jens Tschirpig, <notifications@github.com> wrote:

> *@jens1101* requested changes on this pull request.
>
>    - mysql and pug are not used anywhere in this code. Why are they
>    dependencies?
>    - When I run npm start I receive the following error:
>
>    > moody-socials-backend@1.0.0 start /Users/jens/Documents/Projects/moody-socials/backend
>    > cd server && nodemon server.js
>
>    sh: line 0: cd: server: No such file or directory
>    npm ERR! code ELIFECYCLE
>    npm ERR! errno 1
>    npm ERR! moody-socials-backend@1.0.0 start: `cd server && nodemon server.js`
>    npm ERR! Exit status 1
>    npm ERR!
>    npm ERR! Failed at the moody-socials-backend@1.0.0 start script.
>    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
>
>    npm ERR! A complete log of this run can be found in:
>    npm ERR!     /Users/jens/.npm/_logs/2021-01-27T11_47_30_747Z-debug.log
>
>    Process finished with exit code 1
>
>
> ------------------------------
>
> In backend/routes/pages.js
> <https://github.com/jens1101/moody-socials/pull/2#discussion_r565248153>:
>
> > +    let user = req.session.user;
> +
> +    if(user) {
> +        res.redirect('/home');
> +        return;
> +    }
> +
> +    res.render('index');
> +})
> +
> +
> +router.get('/home', (req, res, next) => {
> +    let user = req.session.user;
> +
> +    if(user) {
> +        res.render('home', {opp:req.session.opp, name:user.username});
>
>
>    - Is a render necessary here? I believe you could just respond with
>    JSON here.
>    - I cannot find any view named "home" in you code, so I don't believe
>    this render will work at all.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jens1101/moody-socials/pull/2#pullrequestreview-577257133>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFQNGY3YRTET754BXFQ2WK3S375AVANCNFSM4WU4FKFA>
> .
>

jens1101 reviewed (changes requested):
Any endpoint that I call responds with a "404 Page not found". Screenshot:

![image](https://user-images.githubusercontent.com/8155712/106105509-c70f2780-614c-11eb-8bfb-ecfb16e5c47a.png)

I believe this is because your error handling middleware in `server.js` is not correctly implemented.

Please test your changes thoroughly first before requesting another review.

jens1101 committed:
Merge branch 'master' into register_login_logout

jens1101 committed:
[WIP] Fixing conflicting backend code

jens1101 committed:
Code cleanup

- Installed prettier globally in the mono repo
- Installed and ran a global "format" script
- Moved backend source code into a `src/` directory. This is to cater for formatting.
- Started on unifying the routes

jens1101 committed:
Removed unnecessary mysql dependency

jens1101 committed:
Removed unnecessary dependencies

jens1101 committed:
Fixed import paths

jens1101 committed:
Added .env support

jens1101 committed:
Used environment variables instead of hard-coded values

jens1101 committed:
Code cleanup

jens1101 committed:
Fixed cookie max age bug

jens1101 committed:
Refined twitter endpoints

jens1101 committed:
Dummy commit

jens1101 committed:
Renamed router
